sdk/python/feast/infra/key_encoding_utils.py

```diff
@@ -6,15 +6,17 @@
 from feast.protos.feast.types.Value_pb2 import ValueType
 
 
-def _serialize_val(value_type, v: ValueProto) -> Tuple[bytes, int]:
+def _serialize_val(
+    value_type, v: ValueProto, entity_key_serialization_version=1
+) -> Tuple[bytes, int]:
     if value_type == "string_val":
         return v.string_val.encode("utf8"), ValueType.STRING
     elif value_type == "bytes_val":
         return v.bytes_val, ValueType.BYTES
     elif value_type == "int32_val":
         return struct.pack("<i", v.int32_val), ValueType.INT32
     elif value_type == "int64_val":
-        return struct.pack("<l", v.int64_val), ValueType.INT64
+        return struct.pack("<q", v.int64_val), ValueType.INT64
```

Comment:
not sure I understand the compatibility approach you're taking here - can you explain?

what do the various values of `entity_key_serialization_version` represent? and there needs to be some fork in the behavior here depending on `entity_key_serialization_version` right?

I was imagining we should do something like the following:
* add `entity_key_serialization_version` to the protos, as you've done, where 0 will represent the current (broken) serialization method and 1 will represent the new (correct) serialization method
* ensure that existing feature view protos do not magically get `entity_key_serialization_version` switched from 0 (the default value since the field doesn't exist) to 1, which requires special logic in the registry (as I think you've done)
* default all new feature views to use serialization method 1